boards/arm/nrf52840dongle_mdk/nrf52840dongle_mdk.dts

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2022 Nikola Trifunovic
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <nordic/nrf52840_qiaa.dtsi>
+
+/ {
+	model = "Nordic nRF52840 Dongle MDK";
+	compatible = "nordic,nrf52840-dongle-mdk";
+
+	chosen {
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,uart-mcumgr = &uart0;
+		zephyr,bt-mon-uart = &uart0;
+		zephyr,bt-c2h-uart = &uart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0_red: led_0 {
+			gpios = <&gpio0 23 GPIO_ACTIVE_LOW>;
+			label = "Red LED 0";
+		};
+		led0_green: led_1 {
+			gpios = <&gpio0 22 GPIO_ACTIVE_LOW>;
+			label = "Green LED 0";
+		};
+		led0_blue: led_2 {
+			gpios = <&gpio0 24 GPIO_ACTIVE_LOW>;
+			label = "Blue LED 0";
+		};		
+	};
+
+	pwmleds {
+		compatible = "pwm-leds";
+		red_pwm_led: pwm_led_0 {
+			pwms = <&pwm0 23>;
+		};
+		green_pwm_led: pwm_led_1 {
+			pwms = <&pwm0 22>;
+		};
+		blue_pwm_led: pwm_led_2 {
+			pwms = <&pwm0 24>;
+		};
+	};
+
+	buttons {
+		compatible = "gpio-keys";
+		button0: button_0 {
+			gpios = <&gpio0 18 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "Push button switch 0";
+		};
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		sw0 = &button0;
+		led0 = &led0_red;
+		led1 = &led0_green;
+		led2 = &led0_blue;
+		led0-red   = &led0_red;
+		led0-green = &led0_green;
+		led0-blue  = &led0_blue;
+		pwm-led0 = &red_pwm_led;
+		pwm-led1 = &green_pwm_led;
+		pwm-led2 = &blue_pwm_led;
+		red-pwm-led = &red_pwm_led;
+		green-pwm-led = &green_pwm_led;
+		blue-pwm-led = &blue_pwm_led;
+	};
+};
+
+&adc {
+	status = "disabled";
+};
+
+&gpiote {
+	status = "okay";
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&uart0 {
+	compatible = "nordic,nrf-uarte";
+	status = "disabled";
+};
+
+&i2c0 {
+	compatible = "nordic,nrf-twi";
+	status = "disabled";
+};
+
+&i2c1 {
+	compatible = "nordic,nrf-twi";
+	/* Cannot be used together with spi1. */
+	status = "disabled";
+};
+
+&pwm0 {
+	status = "okay";
+	ch0-pin = <23>;
+	ch0-inverted;
+	ch1-pin = <22>;
+	ch1-inverted;
+	ch2-pin = <24>;
+	ch2-inverted;
+};
+
+/*
+ * By default, not adding all available SPI instances (spi2, spi3) due to
+ * limited GPIOs available on dongle board.
+ */
+&spi0 {
+	compatible = "nordic,nrf-spi";
+	/* Cannot be used together with i2c0. */
+	status = "disabled";
+};
+
+&spi1 {
+	compatible = "nordic,nrf-spi";
+	status = "disabled";
+};
```

Comment:
note: in general it is safe to set status as `okay`, since it will only have a real effect if enabling the driver. For example, if `uart0` is `okay` but `CONFIG_SERIAL=n` then nothing will happen in practice. If you're doing that to set the compatible you may skip the status property. @anangl can you review this?

Reply:
Setting the status to "okay" would require assigning required pins to those nodes and I guess the intention was to avoid this. I don't see much value in just setting compatibles for the nodes, so I'd rather remove this stuff. However, I think that `uart0` should be set as "okay", since it is referenced by several properties in the `chosen` node.

Reply:
Yes, Andrzej, you are right. I did not set it to OK, so that I did not have to decide which pins should be used. Because the board does not have enough available pins. But I wanted that all available modules are written in the device tree. So that a developer knows what modules are available.
But I can delete them, if this is the usual why.